.chloggen/bc--optimizesnappybufferallocation.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: exporter/prometheusremotewrite
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Remove unused buffer allocation from prometheus remote write execute function
```

Comment:
Should we frame this more as an enhancement? Something like: "Reduce unnecessary memory allocation by removing buffer that was not used by Snappy encoding function."